spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/AsyncCassandraTemplate.java

```diff
@@ -482,7 +488,7 @@ private CqlIdentifier getTableName(Class<?> entityClass) {
 
 		Assert.notNull(entityClass, "Entity type must not be null");
 
-		return doCount(Query.empty(), entityClass, getTableName(entityClass));
+		return doCount(Query.empty(), entityClass, getKeyspace(entityClass), getTableName(entityClass));
```

Comment:
Since the table name and keyspace name are used together, it would make sense to introduce a value object that encapsulates both pieces of information. How about introducing an object `TableName` that contains the table name and an optional keyspace name?

Reply:
I see two propositions from you: 
- create a value object
- leave it as it is, but rename `getKeysapce` to `getKeyspaceName`
So which one should I pick?

Reply:
Let's go with `getKeyspaceName` to align with `getTableName`.

Reply:
So regarding value objects, do you mean that it should be encapsulated and used internally? I.e. passed to the `doCount` whereas the public API should have `getKeyspaceName` and `getTableName`, right?